### PR TITLE
feat: Add period from date range component

### PR DESCRIPTION
**Pull request by johevemi**
## Description
Frontend counterpart https://github.com/Energinet-DataHub/greenforce-frontend/pull/873

## References

* #234

### CODE REVIEW

File: source/dotnet/Domain/BatchAggregate/Batch.cs [hunk 1]
```diff
@@ -22,7 +22,7 @@ public class Batch
 {
     private readonly List<GridAreaCode> _gridAreaCodes;
 
-    public Batch(ProcessType processType, IEnumerable<GridAreaCode> gridAreaCodes)
+    public Batch(ProcessType processType, IEnumerable<GridAreaCode> gridAreaCodes, Instant periodStart, Instant periodEnd)
```
Comment (lasrinnil) [inline source/dotnet/Domain/BatchAggregate/Batch.cs:25]: Move the Create method to a new BatchFactory

File: source/dotnet/DatabaseMigration/Scripts/202209071400_Add_Batch_Period.sql [hunk 2]
```diff
@@ -0,0 +1,4 @@
+﻿ALTER TABLE Batch
+    ADD PeriodStart DATETIME2;
+ALTER TABLE Batch
+    ADD PeriodEnd DATETIME2;
```
Comment (lasrinnil) [inline source/dotnet/DatabaseMigration/Scripts/202209071400_Add_Batch_Period.sql:4]:
```suggestion
﻿ALTER TABLE Batch
    ADD PeriodStart DATETIME2;
ALTER TABLE Batch
    ADD PeriodEnd DATETIME2;
GO
```

File: source/dotnet/Domain/BatchAggregate/Batch.cs [hunk 3]
```diff
@@ -59,7 +57,9 @@ private Batch()
 
     public JobRunId? RunId { get; private set; }
 
-    public Interval Period { get; }
+    public Instant PeriodStart { get; }
```
Comment (lasrinnil) [inline source/dotnet/Domain/BatchAggregate/Batch.cs:60]: A cleaner implementation would be to take to two `instants` in the ctor of the `Batch`  but the model itself would hold a `Interval`

File: source/dotnet/IntegrationTests/ProcessManager/StartPendingBatchesEndpointTests.cs [hunk 4]
```diff
@@ -109,7 +110,9 @@ private async Task<BatchId> CreateAndSavePendingBatch(string gridAreaCode)
 
             var pendingBatch = new Batch(
                 ProcessType.BalanceFixing,
-                new[] { new GridAreaCode(gridAreaCode) });
+                new[] { new GridAreaCode(gridAreaCode) },
+                Instant.FromDateTimeOffset(DateTimeOffset.Now),
```
Comment (lasrinnil) [inline source/dotnet/IntegrationTests/ProcessManager/StartPendingBatchesEndpointTests.cs:114]: Instant.FromDateTimeOffset(DateTimeOffset.Now <- consider moving this to a variable and then adding a day or something for the `EndPeriod`

File: source/dotnet/IntegrationTests/WebApi/V1/BatchControllerTests.cs [hunk 5]
```diff
@@ -63,7 +63,9 @@ public async Task CreateAsync_WhenCalled_AlwaysReturnsOk()
         // Arrange
         var batchRequest = new BatchRequestDto(
             WholesaleProcessType.BalanceFixing,
-            new List<string> { "805" });
+            new List<string> { "805" },
+            DateTimeOffset.Now,
```
Comment (lasrinnil) [inline source/dotnet/IntegrationTests/WebApi/V1/BatchControllerTests.cs:67]: See comment above

File: source/dotnet/Tests/Domain/BatchAggregate/BatchBuilder.cs [hunk 6]
```diff
@@ -15,11 +15,15 @@
...
 using Energinet.DataHub.Wholesale.Domain.GridAreaAggregate;
 using Energinet.DataHub.Wholesale.Domain.ProcessAggregate;
+using NodaTime;
 
 namespace Energinet.DataHub.Wholesale.Tests.Domain.BatchAggregate;
 
 public class BatchBuilder
 {
+    private readonly Instant _periodStart = Instant.FromDateTimeOffset(DateTimeOffset.Now);
+    private readonly Instant _periodEnd = Instant.FromDateTimeOffset(DateTimeOffset.Now);
```
Comment (lasrinnil) [inline source/dotnet/Tests/Domain/BatchAggregate/BatchBuilder.cs:25]: Add an interval to `EndPeriod` to give the reader a better understanding of the intended use

**Review by lasrinnil — COMMENTED** [review]
Please reach out to me if you want any further clarification, I am also more than happy to fix some of my comments!

File: source/dotnet/Domain/BatchAggregate/Batch.cs (hunk 3 above)
Comment (johevemi) [reply source/dotnet/Domain/BatchAggregate/Batch.cs]: Created a check that handles this

File: source/dotnet/Domain/BatchAggregate/Batch.cs (hunk 1 above)
Comment (johevemi) [reply source/dotnet/Domain/BatchAggregate/Batch.cs]: not at this time

**Review by ftoft — APPROVED** [review]

**Pull request merged**